### ISSUE TITLE
sridtype not populated correctly

### ISSUE THREAD

**Issue by kbevers** [Kortforsyningen/FIRE-DDL#47]
Der findes SRID i koordinattabellen som ikke kan findes i sridtype-tabellen:

```
SELECT  unique(srid) FROM koordinat WHERE srid NOT IN (SELECT srid FROM sridtype) ORDER by srid;
```

Listen er her.

```
EPSG:4909
EPSG:5317
EPSG:5799
TS:0
TS:2003
TS:2004
TS:2010
TS:2017
TS:2020
TS:2021
TS:2022
TS:2025
TS:2028
TS:2029
TS:2032
TS:2041
TS:2042
TS:2043
TS:2047
TS:81118
```

De tre EPSG-koder skal manuelt indtastes. De resterende TS: SRID'er forstår jeg ikke helt hvorfor ikke er kommet med men der er nok en logisk forklaring på det. Problemet er opstået fordi der ikke er en foreign key constraint der binder de to tabeller sammen. Havde der været det ville dette være opdaget tidligere.

**Comment by busstoptaktik** [comment]
Assigning @nikolajkamstrup as this is (at least partially) a migration thing

**Comment by kbevers** [comment]
Altså, det med EPSG-koderne bør vi i virkeligheden selv tage os af. Det giver samtidig anledning til at få indsat korrekte beskrivelser af de enkelte systemer:

https://github.com/Kortforsyningen/FIRE-DDL/blob/26d5adb37d4cc7402ee9a066732cbd233b77d581/Migration/MakeSRIDType.sql#L24-L51

Den del synes jeg bør assignes @busstoptaktik :-)

**Comment by busstoptaktik** [comment]
> Den del synes jeg bør assignes @busstoptaktik :-)
Done :-)

**Comment by kbevers** [comment]
Det kunne i øvrigt også være rart med en mere sigende beskrivelse af TS-SRID'erne, fx "Lokalt højdesystem for tidsserie over Jessenpunkt xxxxx" fremfor "Ingen beskrivelse".

**Comment by nikolajkamstrup** [comment]
Jeg tilføjer de tre SRID. TS:0 er en datafejl, vil jeg tro. TS:81118 skal undersøges nærmere, men SRIDTyperne er afledt fra tabellen REFNR_IDENT (lavet af Kristian), mens koordinater er afledt af en tidsserietabel, så mon ikke der er noget, der ikke er i synk. De resterende TS:20xx er dannet på samme måde som vi har dannet TS:81xxx SRID, altså på baggrund af Jessen punkterne i tidsserietabel. TS:81xxx dannes fra tabellen ts_dvr90, TS 20xx fra tabellen ts_euref89. 
Mit forslag er følgende: 1) SRIDTyperne (TS) bør dannes fra samme kilde, som koordinaterne - herved undgår vi forskelle. 2) Jeg skal vide om TS:20xx skal medtages eller ej. Afventer feedback

**Comment by kbevers** [comment]
> Jeg skal vide om TS:20xx skal medtages eller ej. Afventer feedback

Det skal de. Jeg har lige kigget lidt i migreringsscriptet og så vidt jeg kan se tildeles TS:2xxx vha jnr_bside. Det er vist ikke korrekt, da det ser ud til at jnr_bside holder styr på koordinater der er beregnet eller indsat på samme tidspunkt, og ikke den enkelte tidsserie for et punkt som jo er det vi vil holde styr på med TS:.
Så, jeg tror mere det er rigtigt at bruge refnr til at gruppere dem efter. Så vidt jeg kan se kan alle refnumre i ts_euref89 mappes til et GNSS-ident, så det kunne være fedt hvis SRID'erne blev TS:<GNSS-station>. Eksempelvis TS:TAR1 for koordinaterne tilhørende refnr=862726. Der er muligvis nogle enkelte refnr som ikke kan mappes til en GNSS ident, i så fald bør anvendes et andet ident (GI-nummer hvis muligt, ellers landsnr).

Som nævnt ovenfor vil det være fint hvis TS:81xxxx-SRID'er kan få beskrivelsen "Lokalt højdesystem for tidsserie over Jessenpunkt xxxxx" og TS:20xx-SRID'er får beskrivelsen "ETRS89 tidsserie over <GNSS-station>"

**Comment by nikolajkamstrup** [comment]
Ok. Forstået. Det giver 161 TS:xxx SRID, hvoraf 7 ikke har et GNSS eller GI - de får dermed landsnr.

**Comment by kbevers** [comment]
> Ok. Forstået. Det giver 161 TS:xxx SRID, hvoraf 7 ikke har et GNSS eller GI - de får dermed landsnr.

Tjek. Hvis der er mellemrum i landsnr vil det være fordelagtigt at få dem erstattet med `_`.

**Comment by nikolajkamstrup** [comment]
Er der frit valg vedr. talværdien i jeres nye ID kolonner fx SRIDTYPE.SRIDID?

**Comment by nikolajkamstrup** [comment]
Got it, mellemrum i landsnr erstattes med _

**Comment by kbevers** [comment]
> Er der frit valg vedr. talværdien i jeres nye ID kolonner fx SRIDTYPE.SRIDID?

Ja, men det bør være sådan at SRIDtyper, punktinfotyper og observationstyper der er skabt af manuelt skrevne INSERT statements har de samme ID'er hver gang migrationsscriptet køres. Hensigten er at skabe konsistens i hardcodede henvisninger (for dem vil der nok være en del af?). For de typer der skabes automatisk må der bare tildeles fortløbne tal i forlængelse af de manuelt indtastede.
